components/chapters/SignUpModal.tsx

```diff
@@ -0,0 +1,118 @@
+'use client'
+
+import Image from 'next/image'
+import { useState } from 'react'
+import Modal from 'react-modal'
+const crypto = require('crypto')
+const Secp256k1 = require('@lionello/secp256k1-js')
+
+export const SignUpModal = (props) => {
+  function generateKeyPair() {
+    const privateKeyBuf = crypto.randomBytes(32)
+    const privateKey = Secp256k1.uint256(privateKeyBuf, 16)
+
+    const publicKey = Secp256k1.generatePublicKeyFromPrivateKeyData(privateKey);
```

Comment:
This isn't really actionable feedback, but I want to make note (really a note to myself) of storing the public key. In chapter 3 I have an open question about manually deriving the public key from the private key. 

Recall that the public key is actually a point on the elliptic curve. Public key = private key * Generator point

I believe for ECDSA signatures, the public key is 1 byte (indicating if the y coordinate is even or odd) plus 32 bytes for the x coordinate in full.

However for Schnorr, I am reading that the public key is just the x coordinate (32 bytes). Someone please correct me if my understanding is wrong!

This all may lead to a slight discrepancy in what happens in lesson 3 and what we have stored for the public key. Anyways this is not really related to PR review, but something I need to keep in mind!

Reply:
Public keys for ECDSA and Schnorr are both an X and a Y coordinate (32 bytes each).
But yes you are correct, in Bitcoin ECDSA public keys are compressed into 33 bytes: the X coordinate and the parity byte (whether Y is odd or not)
For Schnorr the compressed key is 32 bytes, only the X coordinate is kept and the tie-breaker is implicit.

For lesson 3, since we are deriving ECDSA keys, the user should decide what parity byte to use (either `0x02` if Y is even, otherwise `0x03`. Or we let the user generate uncompressed keys.

Reply:
oooo thank you for the clarification! Would there be any reason to let the user generate a Schnorr public key? 

I am leaning towards ending the lesson at the uncompressed keys, just to avoid this altogether :sweat_smile:

Reply:
> Would there be any reason to let the user generate a Schnorr public key?

I would love to support Schnorr but I believe it is much more complex to derive a P2TR address from a Schnorr public key than a legacy P2PKH from an ECDSA public key. So that may not be very fun for the user.

It shouldn't be too much trouble to ask the user which parity byte to use. Maybe allow the user to run `Y % 2` in the terminal to check if Y is odd or not, or let the user decide randomly until the challenge accepts the solution.

Reply:
I think we stick with uncompressed to start and work our way down in complexity.